Sources/BrowserServicesKit/SecureVault/AutofillDatabaseProvider.swift

```diff
@@ -386,6 +387,34 @@ public final class DefaultAutofillDatabaseProvider: GRDBSecureStorageDatabasePro
         return nil
     }
 
+    public func websiteCredentialsForDomain(_ domain: String) throws -> [SecureVaultModels.WebsiteCredentials] {
+        try db.read {
+            try websiteCredentialsForDomain(domain, in: $0)
+        }
+    }
+
+    func websiteCredentialsForDomain(_ domain: String, in database: Database) throws -> [SecureVaultModels.WebsiteCredentials] {
+        let accounts = try SecureVaultModels.WebsiteAccount
+            .filter(SecureVaultModels.WebsiteAccount.Columns.domain.like(domain))
+            .fetchAll(database)
+
+        let accountIDs = accounts.compactMap(\.id)
+
+        var result = [SecureVaultModels.WebsiteCredentials]()
+
+        for accountID in accountIDs {
+            guard let accountIDInt = Int64(accountID) else {
+                continue
+            }
+            guard let credentials = try websiteCredentialsForAccountId(accountIDInt, in: database) else {
```

Comment:
Calling the `websiteCredentialsForAccountId` method in a loop results in multiple SQL queries. Do we think it’s worth defining a new `websiteCredentialsForAccountId` method which fetches multiple rows in a single transaction? e.g something like:

```
SELECT
            \(SecureVaultModels.WebsiteCredentials.Columns.id.name),
            \(SecureVaultModels.WebsiteCredentials.Columns.password.name)
        FROM
            \(SecureVaultModels.WebsiteCredentials.databaseTableName)
        WHERE
            \(SecureVaultModels.WebsiteCredentials.Columns.id.name) IN ?,?,?,?
        """, arguments: StatementArguments(accountIds))
```

Reply:
Unsure if benefits are worth it, but maybe worth considering.

Reply:
Yeah, I'm not sure. I attempted to, but my SQL queries ended up returning garbage and I'm not sure why. I may need a bit of help writing that query.